DXRenderer/HDRImageViewerRenderer.cpp

```diff
@@ -808,14 +808,23 @@ void HDRImageViewerRenderer::ComputeHdrMetadata()
 
     auto ctx = m_deviceResources->GetD2DDeviceContext();
 
+    // Reset DPI to 100% (96.0f) during histogram calculation, D2D complains
+    // when the resolution is high enough (beyond 4K) and DPI scaling is large.
+    ctx->SetDpi(96.0f, 96.0f);
+
     ctx->BeginDraw();
 
     ctx->DrawImage(m_histogramEffect.Get());
 
     // We ignore D2DERR_RECREATE_TARGET here. This error indicates that the device
     // is lost. It will be handled during the next call to Present.
+    //
+    // We also ignore D2DERR_INTERMEDIATE_TOO_LARGE in case D2D complains the
+    // intermediate tile being too large, the picture should show properly, only
+    // MaxCLL estimation won't work (show as N/A.)
     HRESULT hr = ctx->EndDraw();
-    if (hr != D2DERR_RECREATE_TARGET)
+    ctx->SetDpi(m_deviceResources->GetDpi(), m_deviceResources->GetDpi());
+    if (hr != D2DERR_RECREATE_TARGET && hr != D2DERR_INTERMEDIATE_TOO_LARGE)
```

Comment:
Per my comment in the PR can you rebase on my latest change and remove the ignore D2DERR_INTERMEDIATE_TOO_LARGE?
(https://github.com/13thsymphony/HDRImageViewer/commit/1483b3c438a04828cf408bd4206f72e8519d66d4)

Reply:
Let me take a look later today or maybe this weekend. Just traveled back from Bellevue/Redmond.

By the way, which GPU are you using? Mine is RTX A5000 so technically it falls into the category of some decent card.

Reply:
Take your time!

I'm using an RTX 2080....so yours (Ampere) beats mine (Turing) :). And this confuses me as to my inability to repro the issue, I had thought that this was a DX feature level or GPU architecture limitation. I was using the repro image from #47 and rendering to a 4K display (which shouldn't matter) at 300% DPI.